drivers/bus/acpi/main.c

```diff
@@ -243,29 +243,6 @@ ACPIDispatchDeviceControl(
             {
```

Comment:
So, why is there a separate *ASYNC* eval method IOCTL, if the code considers both IOCTLs to be the same?

Reply:
The IOCTL separation seems to be some sort of optimization. I suppose `IOCTL_ACPI_EVAL_METHOD` was designed for the needs of sensors and low-speed devices. `IOCTL_ACPI_ASYNC_EVAL_METHOD` one is used by low-level storage drivers that handle synchronous paging writes. ACPICA evaluates ACPI control methods synchronously. So I forced both IOCTLs to be synchronous anyway.

Reply:
For what it's worth:

https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/acpiioct/ni-acpiioct-ioctl_acpi_eval_method
```
IOCTL_ACPI_EVAL_METHOD IOCTL
Event is set to a pointer to a caller-allocated and initialized event object.
IOCTL_ACPI_EVAL_METHOD can be used only at IRQL < DISPATCH_LEVEL.
Minimum supported client 	Windows 2000 and later versions of Windows.
```

https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/acpiioct/ni-acpiioct-ioctl_acpi_async_eval_method
```
IOCTL_ACPI_ASYNC_EVAL_METHOD IOCTL
Event is set to NULL.
IOCTL_ACPI_ASYNC_EVAL_METHOD can be used only at IRQL <= DISPATCH_LEVEL.
Minimum supported client 	Windows Vista and later versions of Windows.
```

Reply:
Thanks Serge, but since we're also targeting Vista+ for the new features where there isn't a risk of breaking backward compatibility, in principle we can add support for the async ioctl.

Reply:
@disean I think that @darkfire will certainly revisit this, since he wants to support uACPI.

Reply:
> in principle we can add support for the async ioctl.

Yeah, I was simply pointing out differences on MS documentation.

Reply:
@SergeGautherie Thank you, but `IOCTL_ACPI_ASYNC_EVAL_METHOD` was introduced in Windows 2000.